### ISSUE TITLE
Not saving tab groups anymore

### ISSUE THREAD

**Issue by jayanty** [gilgold/tabox#41]
Tabox has been a lifesaver and I've been using it for months now. Recently, I noticed that it doesn't save tab groups anymore and if I update an older collection with saved tab groups, those groups are getting lost. Has there been a change with recent versions and do I need to update some settings to start saving groups again?

**Comment by gilgold** [comment]
Hey @jayanty I did not release any updates since May so there must be something else going on.
Can you try to disable Tabox and then reenable it? (Go to the Extensions page in Chrome and turn off the switch next to it, then turn it back on).

If that does not work, you can try the following:

1. Export all your collections to a file (in the gear menu, select "Export all collections")
2. Remove Tabox from your browser completely
3. Install Tabox again
4. Import your collections from the file you saved

Let me know if that helped.

**Comment by jayanty** [comment]
Hi @gilgold thank you for those recommendations. 
I tried disabling and re-enabling the extension and then went ahead and removed the extension entirely and then installed it again but both times I still see the problem described earlier. 

After I removed the extension and re-installed it, I did not have to import my collections again because I keep Tabox synced with my Google Account. Not sure if that makes a difference. 

Any other things I could try? Is there a debug log somewhere that I could access?

**Comment by gilgold** [comment]
Do you use any security or blocking extensions? If so, try to disable some of them and see if that help. 
In some cases other extension interfere with Tabox, for example the BitDefender extension is known to cause issues.

**Comment by GalaxySnail** [comment]
I have the same problem on chromium `106.0.5249.62` with tabox `3.5.7`.


![](https://user-images.githubusercontent.com/53637421/194875690-85903ea8-26dd-4495-b35c-269aaf053137.png)

![](https://user-images.githubusercontent.com/53637421/194875689-5bd954e3-2030-4ba5-a16c-194a7eecc662.png)

Indention is still there but the group looks missing.

I exported the collection and I found that `"chromeGroups"` is an empty array, maybe there's something wrong.

``` json
{
    "uid": "c2e43963-ed9c-436c-938c-3ea66e3128ce",
    "name": "example",
    "tabs": [
        {
            "active": false,
            "audible": false,
            "autoDiscardable": true,
            "discarded": false,
            "favIconUrl": "https://github.githubassets.com/favicons/favicon.svg",
            "groupId": 336275679,
            "height": 721,
            "highlighted": false,
            "id": 1483279349,
            "incognito": false,
            "index": 0,
            "mutedInfo": {
                "muted": false
            },
            "pinned": false,
            "selected": false,
            "status": "complete",
            "title": "GitHub",
            "url": "https://github.com/",
            "width": 1535,
            "windowId": 1483279350,
            "uid": "122edd63-194d-43e3-8b1d-0d9ff2af0534"
        },
        {
            "active": false,
            "audible": false,
            "autoDiscardable": true,
            "discarded": false,
            "favIconUrl": "https://cdn.sstatic.net/Sites/stackoverflow/Img/favicon.ico?v=ec617d715196",
            "groupId": 336275679,
            "height": 721,
            "highlighted": false,
            "id": 1483279351,
            "incognito": false,
            "index": 1,
            "mutedInfo": {
                "muted": false
            },
            "pinned": false,
            "selected": false,
            "status": "complete",
            "title": "Stack Overflow - Where Developers Learn, Share, & Build Careers",
            "url": "https://stackoverflow.com/",
            "width": 1535,
            "windowId": 1483279350,
            "uid": "526cbeb6-89b2-4057-b8b6-dc022e07f786"
        },
        {
            "active": true,
            "audible": false,
            "autoDiscardable": true,
            "discarded": false,
            "favIconUrl": "https://static.wixstatic.com/media/8734ed_06d78ef8725b4ba29b57cef24770ecc5~mv2.png/v1/fill/w_32%2Ch_32%2Clg_1%2Cusm_0.66_1.00_0.01/8734ed_06d78ef8725b4ba29b57cef24770ecc5~mv2.png",
            "groupId": -1,
            "height": 721,
            "highlighted": true,
            "id": 1483279357,
            "incognito": false,
            "index": 2,
            "mutedInfo": {
                "muted": false
            },
            "openerTabId": 1483279349,
            "pinned": false,
            "selected": true,
            "status": "complete",
            "title": "Tabox | Save and share tab groups",
            "url": "https://www.tabox.co/",
            "width": 1535,
            "windowId": 1483279350,
            "uid": "bb5c64fc-624c-4dfb-8cb2-6895bf8fa1cb"
        }
    ],
    "chromeGroups": [],
    "color": "var(--bg-color)",
    "createdOn": 1665408135264,
    "window": {
        "alwaysOnTop": false,
        "focused": false,
        "height": 840,
        "id": 1483279350,
        "incognito": false,
        "left": -8,
        "state": "maximized",
        "top": -8,
        "type": "normal",
        "width": 1552
    }
}
```

**Comment by gilgold** [comment]
Hey @GalaxySnail I tried to reproduce it, but it seems to work ok for me, can you try updating Chrome to the latest version (106.0.5249.103) and try again?
Also, please see my previous comment about any other extensions that may interfere. 
Thanks!

**Comment by GalaxySnail** [comment]
Thanks for the quick reply!

It seems to be caused by workona (another browser tab manager). After workona is disabled, tabox works fine for me.

**Comment by gilgold** [comment]
Good to know! Thanks for checking, I will try and see why that extension is causing issues.

**Comment by GalaxySnail** [comment]
Hmmmm I'm sorry but thing are going to be weird. I disabled workona and restarted the browser, at first everything was fine, but after a few tens of minutes later I found the same problem appeared again. Restarting tabox itself didn't work, I had to restart the browser and then tabox gets normal again. I have no idea with it.

**Comment by gilgold** [comment]
Interesting. I will continue to look into it. Thanks for the update!

**Comment by jayanty** [comment]
Discovered something by accident today. I have a few windows which are essentially Chrome Apps (Google Calendar running as a dedicated app for example). With any of those open, groups are not saved. If I close those down, groups are saved. Hopefully @gilgold can replicate that.

**Comment by gilgold** [comment]
@jayanty by 'Chrome Apps' do you mean Chrome OS apps? I am not sure how to have apps run as tabs in the browser on Win or Mac

**Comment by gilgold** [comment]
Ok I. was able to get a hold of a chromebook to test this out, and it seems @jayanty is right, when there is a chrome app open somewhere in the background, groups are not saved. I will dig into this a bit more and hopefully I can add a fix in v3.5.8

Thanks!

**Comment by gilgold** [comment]
I identified the cause and have applied a fix in v3.5.8, it will be released to the Chrome Web Store on Monday.

**Comment by jayanty** [comment]
Wow, that's awesome. Looking forward to trying out the fix. For future reference, here's how you can create dedicated chrome apps on Mac OS:

1. Navigate Chrome to the site of your choice. I do this for Google calendar (https://calendar.google.com)
2. Create Chrome App shortcut from the URL - Chrome drop down menu (3 dots icon on top right) > More Tools > Create Shortcut
3. Check the "Open as window" box in the dialog that appears

Your Google Calendar App is now available as a standalone window app from the MacOS Launchpad.

Instructions borrowed from [here](https://apple.stackexchange.com/questions/236444/is-there-a-google-calendar-mac-desktop-app)

**Comment by gilgold** [comment]
Fixed in v3.5.8 - pending publish in the Chrome and Edge stores
